### ISSUE TITLE
renormalization of state reward 

### ISSUE THREAD

**Issue by waynezw0618** [tensorforce/tensorforce#720]
Hello Alex

do we need to renormalize the state reward ? I found the reward keep accumulating, which leads to the divergency of the training. But we can not limit the accumulated value with a certain function for the reward or state.?

**Comment by AlexKuhnle** [comment]
Hi,

Can you give more detail on your training setup: What agent config? Episode length? Reward value range? Also, how do you find out rewards keep accumulating?

Generally, it's recommended to have the reward values center around zero and not become too big, basically more or less normally distributed. If episodes are long and reward signals dense (so there are rewards more or less at every timestep), the discount value can be very important, since it prevents the cumulative return from getting too big.

**Comment by waynezw0618** [comment]
for a single step the reward value is ranging from 0 to 1, which is of function exp(-abs(x)). the Max Episode length is 20000, but there is a definition of reward for the last step which is either 1(succeed) or -50(failed)..
I am using PPO with Adam method as the optimiser and the  learning_rate is set to 1e-3,

**Comment by AlexKuhnle** [comment]
Are you setting a discount value (`!= 1.0`)? Sounds like this would definitely be a good idea. Can you also talk more about how you want to renormalize the state reward, and why? And what do you mean by "the reward keeps accumulating"?

**Comment by waynezw0618** [comment]
https://github.com/waynezw0618/boatNav/blob/c19874c9093d5060938ba5395b562dbf742f495f/src/boat_env/boat.py#L177:202

**Comment by waynezw0618** [comment]
https://github.com/waynezw0618/boatNav/blob/c19874c9093d5060938ba5395b562dbf742f495f/src/boat_env/boat.py#L177

**Comment by waynezw0618** [comment]
Hi Alex
I invited you to my repo. would you please take a look at ?

**Comment by AlexKuhnle** [comment]
Okay, will do.

**Comment by AlexKuhnle** [comment]
I assume this can be closed.
